url/url.go

```
@@ -0,0 +1,326 @@
+//
+//	To construct a URL with query parameters:
+//	  queryParams := map[string]string{"key": "value", "anotherKey": "anotherValue"}
+//	  url, err := BuildURL("http", "example.com", "search", queryParams)
+//	  fmt.Println("Constructed URL:", url)
+func BuildURL(scheme, host, path string, query map[string]string) (string, error) {
+	var errMessage []string
+	if scheme == "" {
+		errMessage = append(errMessage, "scheme is required")
+	}
+	if host != "" {
+		re := regexp.MustCompile("^[a-zA-Z0-9-]+(\\.[a-zA-Z0-9-]+)+$")
```

golangci-lint

S1007: should use raw string (`...`) with regexp.MustCompile to avoid having to escape twice

(gosimple)